src/revwalk.rs

```diff
@@ -119,6 +133,20 @@ impl<'repo> Revwalk<'repo> {
         Ok(())
     }
 
+    /// Hide all commits for which the callback returns true from
+    /// the walk.
+    pub fn hide_callback(&mut self, callback: Box<dyn FnMut(Oid) -> bool>) -> Result<(), Error> {
```

Comment:
Ideally this would take `impl FnMut(...)` rather than a `Box` (but we'd probably box it up intternally). Additionally this would ideally not have a `'static` bound similar to other callbacks in these bindings. Adding more lifetime arguments would be a breaking change to `Revwalk`, but we could at least just bound the callback by `'repo` I believe.

Reply:
I tried to do do it that way (boxing internally) but could not get it to compile. It suggests me to add `'static` bound which is not desirable. Not sure `'repo` is right either because the lifetime of both the `Revwalk` as well as the callback can be much shorter than `'repo`.

Reply:
I tried: 
```suggestion
        pub fn hide_callback<C: 'repo>(&mut self, callback: C) -> Result<(), Error> 
    where
        C: FnMut(Oid) -> bool,

```
But that does not compile :(

Reply:
Hm ok, well in any case relaxing the lifetime bound here will require adjusting the parameters on `Revwalk`. For now it's fine to just require `'static` and we can fix this in a later breaking change if necessary. Can this still switch to the generic version with a `'static` bound though rather than taking a trait object here?

Reply:
So I came up with a different way to introduce this API that is quite I bit better I believe. (See new commit)
Comment and docs will still need to be updated if you agree with the approach taken.